### ISSUE TITLE
BerquistSherman class paid adjustment causes NaN's[BUG]

### ISSUE THREAD

**Issue by cdietrich215** [casact/chainladder-python#537]
**Describe the bug**
I believe the BerquistSherman class's paid adjustment has a bug (this probably some lesser used functionality again). This error occurs in the [lookup array](https://github.com/casact/chainladder-python/blob/master/chainladder/adjustments/berqsherm.py#L127) defined in line 127. 

The procedure selects regression parameters (`a` and `b`) within the same AY, based on how close the adjusted closed claim count is to the raw closed claim counts in the row, as described in the Friedland paper, page 289. However, in this implementation, if the adjusted closed count is higher than the max of the AY row's unadjusted closed counts, this lookup produces a value that extends into the "lower triangle". This causes `nan` values for the `a` and `b` parameters, as well as the resulting paid loss values. 

There should be a ceiling on the lookup values corresponding to the final regression in the AY row. This can be done after lookup is defined  in line 143:

```
n = lookup.shape[-1]
for j in range(n - 1):
    lookup[:, :, j, :] = np.clip(lookup[:, :, :, j], 0,  n - j - 1)
```

Hope this makes sense - will try to produce an example at some point. Happy to help implement this.

**Comment by cdietrich215** [comment]
@jbogaardt I would be happy to help contribute on this or some of the other issues I brought up, let me know if that would be an option!

**Comment by jbogaardt** [comment]
@cdietrich215 , contributions are very welcome! Please contribute if you can! Any PR will be tested against the existing test suite. If you can keep those in a passing state, things should be good. Happy to provide further pointers on setting up a development environment (or alternatively you can reference https://chainladder-python.readthedocs.io/en/latest/library/contributing.html)

**Comment by cdietrich215** [comment]
Sounds good, was able to get the environment set up. Will make a PR on this soon.

**Comment by cdietrich215** [comment]
@jbogaardt Do I need additional permissions to do so? I'm getting permission denied errors when attempting to push a new branch: `remote: Permission to casact/chainladder-python.git denied to cdietrich215.`

**Comment by cdietrich215** [comment]
Additionally, this will replicate the error mentioned above. The code in the feature branch I'll push should prevent the issue.

```import chainladder as cl
triangle = cl.load_sample("berqsherm").loc['MedMal']
xp = triangle.get_array_module()
# Purposefully cause negative closed development towards the end for demonstration
triangle.loc["MedMal", "Closed", "1969" , 84] = triangle.loc["MedMal", "Closed", "1969" , 72] * 0.9
triangle.loc["MedMal", "Closed", "1969" , 96] = triangle.loc["MedMal", "Closed", "1969" , 84] * 0.9
triangle.loc["MedMal", "Closed", "1970" , 84] = triangle.loc["MedMal", "Closed", "1970" , 72] * 0.9

berq = cl.BerquistSherman(
    paid_amount='Paid', incurred_amount='Incurred',
    reported_count='Reported', closed_count='Closed',
    trend=0.15)
# Transform triangle
berq_triangle = berq.fit_transform(triangle)```

**Comment by jbogaardt** [comment]
Hi @cdietrich215,

You wouldn't be able to push anything directly to casact/chainladder-python without being a project maintainer. However, this is easy to resolve.

To submit a pull request (PR) to any public repository on GitHub, fork the repository, create a new branch on your fork, make your changes, commit them, push the changes to your fork, then navigate to the repository on GitHub and initiate a pull request by selecting the "Compare & pull request" option, choosing the appropriate branches, and providing a descriptive title and explanation for your changes; the repository maintainers will then review your PR and decide whether to merge your changes into the main codebase.
